bokeh/sphinxext/bokeh_gallery.py

```diff
@@ -76,15 +83,23 @@ def run(self):
             raise SphinxError(f"gallery dir {gallery_dir!r} missing for gallery file {gallery_file!r}")
 
         spec = json.load(open(gallery_file))
-        names = [detail["name"] for detail in spec["details"]]
+        names = [detail["path"] for detail in spec["details"]]
+
+        opts = []
+        for example in names:
+            pp = PurePath(example).parts
+            name = pp[-1].replace('.py', '')
+            opts.append({"name": name, "ref": f'examples/{pp[1]}/{name}.html'})
```

Comment:
As you can see, I only use the path to the example. To make this possible I changed the name of `pie_chart.png` to `pie.png`. This was the only file, there the name was drifferent to the file name. This means, that the json-file can be removed, if we can store a list of file paths somewhere. I first thought it can go to `conf.py` but I don't know how to acces this list inside a `Directive`.

I think there will be some leftovers which should be removed, too. There I need some help.